File: python/leap/leap.py
def leap_year(year: int) -> bool:
    """
    Function to return boolean value if provided year is a leap year

    :param year: int -the provided year
    :return: int - whether provided year is leap year or not
    """
    if year % 400 == 0:
        return True

    return year % 4 == 0 and year % 100 != 0

File: zig/leap/leap.zig
// pub fn isLeapYear(year: u32) bool {
//     if (year % 4 == 0) {
//         if (year % 100 == 0) {
//             if (year % 400 == 0) {
//                 return true;
//             }
//             return false;
//         }
//         return true;
//     }
//     return false;
// }

pub fn isLeapYear(year: u32) bool {
    if (year % 4 == 0) {
        if (year % 100 == 0) {
            return if (year % 400 == 0) true else false;
        } else {
            return true;
        }
    }

    return false;
}
